Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -1478,14 +1464,16 @@ void LeBailFit::execRandomWalkMinimizer(size_t maxcycles,
   Rfactor startR(-DBL_MAX, -DBL_MAX);
 
   // Process background to make a pure peak spectrum in output workspace
-  MantidVec &vecBkgd = m_outputWS->dataY(INPUTBKGDINDEX);
-  m_lebailFunction->function(vecBkgd, vecX, false, true);
-  MantidVec &dataPurePeak = m_outputWS->dataY(INPUTPUREPEAKINDEX);
+  std::vector<double> vecBkgd =
+      m_lebailFunction->function(vecX.rawData(), false, true);
+  m_outputWS->mutableY(INPUTBKGDINDEX) = vecBkgd;
+  std::vector<double> dataPurePeak(m_outputWS->y(INPUTPUREPEAKINDEX).size(), 0);
   transform(vecInY.begin(), vecInY.end(), vecBkgd.begin(), dataPurePeak.begin(),
             ::minus<double>());
```

Comment:
Why not store the result of `m_lebailFunction->function` in a `HistogramY` (or directly return one?)? THen you can simply use `operator-` or `-=` here.